### ISSUE TITLE
opening a file in a new window is unnecessarily difficult

### ISSUE THREAD

**Issue by PlasticToad22** [atom/atom#13558]
### Prerequisites

* [ X ] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

This has been tested on macOS and Windows, with macOS being the target platform.

Suppose we want to use two Atom windows on a regular basis. The window opened second seems to be a second-class citizen and does not behave as a user of other editors (e.g. Sublime, TextWrangler, Notepad++, etc) might expect.

For instance, if you drag-and-drop a file from Finder into Atom, it opens in a new tab. This is great, except if we drag to the Secondary window. This will open the file in the Primary window, which is Not Good. This scenario is our main issue, however there are two more issues which exacerbate the problem:

- One is unable to open the Tree View in the Secondary window. Selecting the menu option or using the keyboard shortcut does nothing. So, the Main issue cannot be circumvented in this way.

- Dragging a tab from the Primary window to the Secondary window creates a duplicate of the tab. This would be neat if both buffers were updated simultaneously, but since they are not, it ends up being confusing, and therefore results in the extra step of having to close the first tab.

### Steps to Reproduce

1. Open the Atom editor
2. Open a New Window (you will now have two distinct windows)
3. Drag a file into the window created second
4. The file will appear in the first window

**Expected behavior:** The file should appear in the window targeted by the user, and nowhere else.

**Actual behavior:** The file appears in the first window created, no matter what. Other issues, as described above, make this more annoying than it might otherwise be.

**Reproduces how often:** 100% - Always

### Versions

macOS 10.12.2 - Atom 1.12.8
Windows 10 v1607 - Atom 1.12.9

### Additional Information

This can be reproduced on a clean install of Atom, with no additional configuration or packages.

**Comment by 50Wliu** [comment]
I cannot confirm the following:
> One is unable to open the Tree View in the Secondary window. Selecting the menu option or using the keyboard shortcut does nothing. So, the Main issue cannot be circumvented in this way.

I was able to open the Tree View by adding a project folder to the new window, and then toggling the tree view.

> For instance, if you drag-and-drop a file from Finder into Atom, it opens in a new tab. This is great, except if we drag to the Secondary window. This will open the file in the Primary window, which is Not Good.

I cannot reproduce this using Windows Explorer on Windows 10 x64.  The file opens in the secondary window as expected.

This is using 1.14.0-dev-fa1293e (master merged with Electron 1.4 branch).

**Comment by PlasticToad22** [comment]
Thanks for your response. First, I must note that while the macOS version I'm using on my primary Mac is 1.12.8, it says "Atom is up to date!" when I check the version in GUI, which I guess could be considered buggy but I'm not worried about it personally, and I don't know if you're even checking secondary-point releases. I was/am using 1.12.9 on the other two Macs I tested on (all 2015 iMacs). I updated to 1.12.9 on the primary Mac, no luck. I then tried building the latest Atom with Electron 1.4, as suggested:

> Atom    : 1.14.0-dev-09ab7d6
> Electron: 1.4.13
> Chrome  : 53.0.2785.143
> Node    : 6.5.0

...but unfortunately did not get the desired results. I also tried out the 1.13.0-beta10 of Atom on Windows 10, with no luck. (I do not want to build the latest on Windows 10 because I do not enjoy using Windows.)

However, I may have found the problem. While trying a few things, I ⌘-O'd a file instead of dragging it or using the Tree View. If the Secondary window is in focus, this suddenly fixes the problem, permanently – dragging-and-dropping works correctly, and Tree View shows up in the Secondary window. I guessed then that it must have something to do with the post-install configuration. Sure enough, if you delete `~/.atom/storage/application.json` the problem returns. It looks like, on a fresh install, that file will look like this: `{"initialPaths":[]}`, with the brackets being filled with the absolute path of the first open file. However, the Secondary window never receives this state if you do not ⌘-O while having it in focus; it always remains `{"initialPaths":[]}`, which gives Atom no frame of reference for Tree View, and presumably some other weirdness which doesn't allow drag-and-drop. All of this behavior was reproduced on Windows.

Therefore, to reproduce the problem, start with no Atom config directory, or at least delete/move storage/application.json from your config profile.

I suppose this could be considered fixed since it now works for me, but it was very off-putting as a new Atom user. If my experience can be verified, it would probably be better to have different behavior from the outset.

**Comment by PlasticToad22** [comment]
Addendum: If the path filled in `{"initialPaths":[]}` is different from the path of the file one attempts to drag-and-drop into the Second window of Atom, the buggy behavior returns, for me anyway. A friend can reproduce the above comment, but not this footnote.

Also, with the latest build and Electron 1.4, I can reproduce the issue, but not the fix, in the above comment, which is particularly bizarre. Even if I `vim ~/.atom/storage/applications.json` and manually add a path for the Secondary window, Atom simply sets it back to blank on the drag-and-drop attempt.

**Comment by PlasticToad22** [comment]
I created a virgin macOS 10.12.2 virtual machine and installed the latest Atom (1.13.0), and it performs exactly as described here, so literally anyone can reproduce the problem.

**Comment by ungb** [comment]
I am able to repro the issue from the comment here:  https://github.com/atom/atom/issues/13558#issuecomment-271012364

If you open a file that isn't part of either of the directories currently open, it seems to work correctly.

Repro steps(have two folder,  Lets call it `Folder A` and `Folder B`)
1. Open `Folder A` with Atom
2. Open a new window of Atom with `Folder B`
3. Open file explorer/finder, and drag a file from `Folder A` into the atom window where `Folder B` is open

Expected:  file should be open at the Atom window it was drag to,

Actual: File is opened in the Atom window that has `Folder A` opened at.  


NOTE: if you try dragging a file from a folder that isn't loaded on atom, this works correctly.

**Comment by MichaelEischer** [comment]
I'm also able to reproduce the drag&drop issue, which seems to be similar to #10283 .
I use several atom windows which are spread over multiple virtual desktops. This makes opening a file in the wrong editor window especially annoying as atom switches to a different desktop and opens the file there.
Thank you for the reproduction steps, now I'm at least able to work around the problem.

**Comment by 50Wliu** [comment]
@connorbrinton can you open a new issue for the first one on https://github.com/atom/tabs?

**Comment by connorbrinton** [comment]
@50Wliu Yup! I found a [similar issue](https://github.com/atom/tabs/issues/141) there and added a comment. Thanks for pointing me in that direction 😁
